lib/Server.js

```diff
@@ -323,7 +323,9 @@ function Server(compiler, options) {
 
     middleware: () => {
       // include our middleware to ensure it is able to handle '/index.html' request after redirect
+      if (typeof options.before === 'function') { options.before(app, this); }
```

Comment:
preferably we'd follow the same pattern established on line 342 with the `setup` middleware. `'before'` would be inserted into that array before `setup` and `after` would be added to the array before line 355

Reply:
Fixed!
I've changed the example for middleware as well, saw that one was using the `setup` hook which is now the `before` hook.